apps/wallet-dashboard/tests/stake.spec.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { test, expect } from './fixtures';
+import { connectWallet, requestFaucetTokensOnWalletHome } from './utils';
+import 'dotenv/config';
+
+test.describe('Wallet staking', () => {
+    test.setTimeout(30_000);
+
+    test('should allow to stake and unstake funds', async ({
+        context,
+        pageWithFreshWallet,
+        extensionName,
+    }) => {
+        await pageWithFreshWallet.bringToFront();
+        await requestFaucetTokensOnWalletHome(pageWithFreshWallet);
+
+        const dashboardPage = await context.newPage();
+        await dashboardPage.goto('/');
+        await connectWallet(dashboardPage, context, extensionName);
+
+        await dashboardPage.getByTestId('sidebar-staking').click();
+        await dashboardPage.getByRole('button', { name: 'Stake' }).click();
+
+        await dashboardPage.getByText('validator-1').click();
+        await dashboardPage.getByText('Next').click();
+
+        await dashboardPage.getByLabel('Amount').fill('10');
+
+        let stakeButton = dashboardPage.getByTestId('stake-confirm-btn');
+        await expect(stakeButton).toBeVisible();
+
+        stakeButton = dashboardPage.getByTestId('stake-confirm-btn');
+        let walletApprovePagePromise = context.waitForEvent('page');
+        await stakeButton.click();
+
+        let walletApprovePage = await walletApprovePagePromise;
+        await walletApprovePage.getByRole('button', { name: 'Approve' }).click();
+
+        await expect(dashboardPage.getByText('Successfully sent')).toBeVisible({
+            timeout: 30_000,
+        });
+
+        await dashboardPage.getByTestId('close-icon').click();
+
+        await dashboardPage.reload();
+
+        const stakedAmount = await dashboardPage
+            .locator('div:has(> span:text("Your stake"))')
+            .locator('xpath=../../div/span')
+            .first()
+            .textContent();
+        expect(stakedAmount).toEqual('10');
+
+        // UNSTAKE
+        await dashboardPage.getByText('validator-1').click();
+        await dashboardPage.getByText('Unstake').click();
+
+        walletApprovePagePromise = context.waitForEvent('page');
+        await dashboardPage.getByRole('button', { name: 'Unstake' }).click();
+        walletApprovePage = await walletApprovePagePromise;
+        await walletApprovePage.getByRole('button', { name: 'Approve' }).click();
+
+        await dashboardPage.reload();
```

Comment:
```suggestion
        await dashboardPage.waitForSelector('text=Start Staking', {
            timeout: 30_000,
        });
```
or
```
await dashboardPage.reload({
            waitUntil: 'load',
        });
```

I'd recommend to wait by text instead of reload page.
I experienced flaky behaviour when page reloaded, test continue work and fail because of validator-1 still on the page (debug mode)
<img width="787" alt="image" src="https://github.com/user-attachments/assets/5382cb57-6050-4ca2-9f6d-9d326743ceab" />

Reply:
Done in [0a32f2b](https://github.com/iotaledger/iota/pull/6489/commits/0a32f2bd4e93811756c401ac1cb3023bfb53d748)